ably/realtime_channel.go

```diff
@@ -50,6 +52,15 @@ type ChannelOption func(*channelOptions)
 // channelOptions wraps ChannelOptions. It exists so that users can't implement their own ChannelOption.
 type channelOptions protoChannelOptions
 
+type DeriveOptions struct {
```

Comment:
You should add some docstrings to this public struct.

In fact, we should really be adding new SDK API descriptions to the [sdk-api-reference](https://github.com/ably/sdk-api-reference/blob/main/descriptions.md) so that they're aligned between SDKs.